### PR TITLE
update API interface

### PR DESCRIPTION
**Pull request by TessFerrandez**
# PR for issue #165 

## What is being addressed

Update the API interface for /workspaces per the description in story #161 

## How is this addressed

- Input parameters for POST simplified to friendlyName, workspaceType and description
- Output parameters for POST simplified to only the ID
- core_id, and location are accessed from environment variables
- Workspaces structure changed to match Porter - Parameters are inside the ResourceSpec and stored as a list (name/value)
- Tests updated
- Examples (that will be shown in the API/docs) - added or updated to reflect the input/output schema

### CODE REVIEW

**Comment by TessFerrandez** [comment]
Will update with documentation before moving to PR

File: management_api_app/tests/test_api/test_routes/test_workspaces.py [hunk 1]
```diff
@@ -80,20 +92,26 @@ async def test_workspaces_id_get_returns_422_if_workspace_id_is_not_a_uuid(get_w
 @patch("api.dependencies.workspaces.WorkspaceRepository.get_workspace_by_workspace_id")
 async def test_workspaces_id_get_returns_workspace_if_found(get_workspace_mock, app: FastAPI, client: AsyncClient):
     sample_workspace = {
-        "id": "afa000d3-82da-4bfc-b6e9-9a7853ef753e",
+        "id": "933ad738-7265-4b5f-9eae-a1a62928772e",
         "resourceSpec": {
-            "name": "tre-workspace-vanilla",
+            "name": "My workspace",
```
Comment (marrobi) [inline management_api_app/tests/test_api/test_routes/test_workspaces.py:97]: Just to make sure I'm not getting confused - this is the bundle name? And friendly name is the workspace name?

Maybe we should have them as `bundleName` and `workspaceName` to be explicit.

**Comment by marrobi** [comment]
Looking good I will build and deploy and try creating and retrieving some workspaces.

File: management_api_app/tests/test_api/test_routes/test_workspaces.py (hunk 1 above)
Comment (TessFerrandez) [reply management_api_app/tests/test_api/test_routes/test_workspaces.py:97]: Good catch - this was a copy paste from the example schema - will change both here and in the schema to tre-workspace-vanilla

File: management_api_app/db/repositories/workspaces.py [hunk 2]
```diff
@@ -34,13 +35,20 @@ def get_workspace_by_workspace_id(self, workspace_id: UUID4) -> Resource:
...
+    def create_workspace(self, workspace_create: WorkspaceInCreate) -> Workspace:
+        workspace = Workspace(
             id=str(uuid.uuid4()),
-            resourceSpec=ResourceSpec(name=workspace_create.resourceSpec.name, version=workspace_create.resourceSpec.version),
-            parameters=workspace_create.parameters,
-            resourceType=ResourceType.Workspace,
-            status=Status.NotDeployed
+            friendlyName=workspace_create.friendlyName,
+            description=workspace_create.description,
+            resourceSpec=ResourceSpec(
```
Comment (sachinkundu) [inline management_api_app/db/repositories/workspaces.py:43]: Hmm the actual spec should have come from the ResourceSpecs container using name and version. Is that done somewhere else or later?

Comment (TessFerrandez) [reply management_api_app/db/repositories/workspaces.py:43]: can you elaborate -- the workspaceType here is coming in from the user as the resourceSpec name (e.g. tre-vanilla-...), the version will be calculated as the latest for that spec in #167

Do you mean the parameters?

It becomes a little difficult as the parameters are both parameters that are fixed - that should be provided by the API
(the user can't give neither the workspace_id, location, core_id or really the address_space.) and potentially some that are not fixed... perhaps we should rethink this and make the ones that should be provided by the API move outside the params section.

File: management_api_app/tests/test_api/test_routes/test_workspaces.py (hunk 1 above)
Comment (sachinkundu) [reply management_api_app/tests/test_api/test_routes/test_workspaces.py:97]: you should read it as ``resourceSpec_name``

File: management_api_app/api/routes/workspaces.py [hunk 3]
```diff
@@ -3,23 +3,23 @@
 
 from api.dependencies.workspaces import get_repository, get_workspace_by_workspace_id_from_path
 from db.repositories.workspaces import WorkspaceRepository
-from models.domain.resource import Resource
-from models.schemas.resource import ResourcesInList, ResourceInResponse, ResourceInCreate
+from models.domain.workspace import Workspace
+from models.schemas.workspace import WorkspaceInCreate, WorkspaceIdInResponse, WorkspacesInList, WorkspaceInResponse
```
Comment (sachinkundu) [inline management_api_app/api/routes/workspaces.py:7]: I am a bit confused with the meaning of schema here. IS this request-response schemas and is it a fast api convention to call these schemas?

**Review by sachinkundu — APPROVED** [review]

File: management_api_app/db/repositories/workspaces.py (hunk 2 above)
Comment (sachinkundu) [reply management_api_app/db/repositories/workspaces.py:43]: I am saying that when creating a workspace you receive name and version of the bundle. We should get the spec(which was pre registered with us) from the cosmos container ResourceSpecs using the given name and version. Then we know what spec to fill and ideally we would validate the presence of input parameters using this spec(but we are not doing that right now) Spec is something to read from the database.

Comment (TessFerrandez) [reply management_api_app/db/repositories/workspaces.py:43]: right... but it gets complicated, let's discuss in standup

File: management_api_app/api/routes/workspaces.py (hunk 3 above)
Comment (TessFerrandez) [reply management_api_app/api/routes/workspaces.py:7]: yes and yes

File: management_api_app/db/repositories/workspaces.py (hunk 2 above)
Comment (TessFerrandez) [reply management_api_app/db/repositories/workspaces.py]: Resolving per offline discussions - moving this to a more flat structure to differentiate between ResourceSpec (the specification) and what we send as input (like in this case)

**Pull request merged**